.github/workflows/publish-docker-tag.yml

```diff
@@ -3,20 +3,38 @@ on:
   push:
     tags:
       - '*'
+  workflow_dispatch:
 
 name: Publish tagged image to Docker Hub
 
 jobs:
-  build:
-    runs-on: ubuntu-18.04
+  docker-tag:
+    runs-on: self-hosted
     steps:
-      - uses: actions/checkout@v2
-      - name: Publish to Registry
-        uses: elgohr/Publish-Docker-Github-Action@master
-        env:
-          COMMIT_SHA: ${{ github.sha }}
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```

Comment:
Here if I understand correctly we setup qemu so we can virtualize “any” platform and build our binary for it?
Is this better than [`cargo-cross`](https://github.com/rust-embedded/cross) which is made specifically to cross compile rust to other platform?

Reply:
I don't know if it's better, but if I understand correctly,  using `cargo-cross` would involve creating another dockerfile or modifying the current one, right?
The CI creates both docker images, and according to the machine of the user, will pull the right one

Reply:
Buildx concurrently builds same image for multiples architecture and that includes a kind of manifest. You are right, when user pulls the image it gets the right one , based on machine architecture. 

I don’t know about cargo-cross. But buildx is the official tool from Docker.

Reply:
Ah ok, from what I see we would need to keep buildx and there is no good integration with cargo-cross so lets call it a day and roll with it 😁